pkg/controller/operators/catalog/operator.go

```diff
@@ -2040,11 +2044,12 @@ func (o *Operator) apiresourceFromGVK(gvk schema.GroupVersionKind) (metav1.APIRe
 	}
 	for _, r := range resources.APIResources {
 		if r.Kind == gvk.Kind {
+			logger.WithField("kind", r.Kind).Infof("found GVK %s in api discovery", gvk.String())
 			return r, nil
 		}
 	}
 	logger.Info("couldn't find GVK in api discovery")
-	return metav1.APIResource{}, olmerrors.GroupVersionKindNotFoundError{gvk.Group, gvk.Version, gvk.Kind}
```

Comment:
Any reason we're not using the `GroupVersionKindNotFound` error?

Reply:
@benluddy objected to the fact that this function returns different error types for no reason, and the caller had to be aware of that fact. I generally agree since this custom error type serves no purpose in our code ie we don't look for it

Reply:
SGTM